plugins/webp-uploads/helper.php

```diff
@@ -106,21 +106,22 @@ function webp_uploads_generate_additional_image_source( int $attachment_id, stri
 	if ( is_wp_error( $image ) ) {
 		return $image;
 	}
+	if ( is_array( $image ) && array_key_exists( 'file', $image ) && is_string( $image['file'] ) ) {
+		// The filtered image provided all we need to short-circuit here.
+		if ( array_key_exists( 'filesize', $image ) && is_int( $image['filesize'] ) && $image['filesize'] > 0 ) {
+			return $image;
+		}
 
-	if (
-		is_array( $image ) &&
-		! empty( $image['file'] ) &&
-		(
-			! empty( $image['path'] ) ||
-			array_key_exists( 'filesize', $image )
-		)
-	) {
-		return array(
-			'file'     => $image['file'],
-			'filesize' => array_key_exists( 'filesize', $image )
-				? $image['filesize']
-				: wp_filesize( $image['path'] ),
-		);
+		// Supply the filesize based on the filter-provided path.
+		if ( array_key_exists( 'path', $image ) && is_int( $image['path'] ) ) {
+			$filesize = wp_filesize( $image['path'] );
+			if ( $filesize > 0 ) {
+				return array(
+					'file'     => $image['file'],
+					'filesize' => $filesize,
+				);
+			}
+		}
```

Comment:
This improved/hardened validation logic resulted in the test change `tests/plugins/webp-uploads/helper-tests.php` wherein the filtered value was pointing to an invalid `path` which was not resulting in an incorrect overridden value. Now that there is validation for the filtered value, the value will only be used if it is valid.